src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -676,11 +676,19 @@ def _addACDCFileToWMBSFile(self, acdcFile, inFileset=True):
         adds the ACDC files into WMBS database
         """
         wmbsParents = []
-        for parent in acdcFile["parents"]:
-            parent = self._addACDCFileToWMBSFile(DatastructFile(lfn=parent,
-                                                                locations=acdcFile["locations"]),
-                                                 inFileset=False)
-            wmbsParents.append(parent)
+        # TODO:  this check can be removed when ErrorHandler filters parents file for unmerged data
+        # If files is merged and has unmerged parents skip the wmbs population
+        if acdcFile.get("merged", 0) and len(acdcFile["parents"]) and ("/store/unmerged/" in next(iter(acdcFile["parents"]))) :
```

Comment:
Same comment about merged. And the above comment is the contrary of this `if` line, did you mean if input is *not* merged, then skip wmbs population (?)